maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java

```diff
@@ -86,6 +86,39 @@ public String getVersion()
         }
     }
 
+    @Override
```

Comment:
On a related note, why is it necessary to override these setters? I can see that in certain conditions, they will create a new instance of the `RelocatedArtifact` class rather than mutating the existing one. In general: yay for immutability!

But the Maven codebase usually assumes a `setX` method to mutate the object it is called upon and ignores the return value (if any). So what problem does it solve, and are we sure that whenever we invoke it, we continue with the new object rather than the (unmodified) original object?

Reply:
Yes, we are using the new object. Checked myself and tripped over. I can drop the `@Override`, no issue.